src/AutoRest.CSharp/MgmtTest/Generation/MgmtTestWriterBase.cs

```diff
@@ -129,12 +129,19 @@ protected CodeWriterDeclaration WriteGetResource(MgmtTypeProvider carrierResourc
 
         protected CodeWriterDeclaration WriteGetFromResource(Resource carrierResource, OperationExample example, FormattableString client)
         {
-            var idVar = new CodeWriterDeclaration($"{carrierResource.Type.Name}Id".ToVariableName());
-            WriteCreateResourceIdentifier(example, idVar, carrierResource.RequestPath, carrierResource.Type);
-            var resourceVar = new CodeWriterDeclaration(carrierResource.ResourceName.ToVariableName());
-            _writer.Line($"{carrierResource.Type} {resourceVar:D} = {client}.Get{carrierResource.Type.Name}({idVar});");
+            if (carrierResource.Type.Name == nameof(TenantResource))
```

Comment:
how about we just use
```suggestion
            if (carrierResource.Type.Equals(typeof(TenantResource)))
```